frontend/src/components/EditableSelect.tsx

```diff
@@ -0,0 +1,151 @@
+  const [editValue, setEditValue] = useState<Option | undefined>(undefined);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleConfirm = () => {
+    setIsLoading(true);
+    onOk(editValue!.value, fieldId).then((returnedError) => {
+      if (returnedError) setErrorMessage(returnedError);
+      else setEditOpen(false);
+      setIsLoading(false);
+    });
+  };
+
+  const handleCancel = () => {
+    setErrorMessage('');
+    setEditValue(undefined);
+    setEditOpen(false);
+  };
+
+  return (
+    <div
+      className={classes(css.withButtonsContainer, {
+        [css.withButtonsContainerEditMode]: editOpen,
+      })}
+    >
+      <Component
+        id={fieldId}
+        editOpen={editOpen}
+        editValue={editValue}
+        onChange={(newValue) => {
+          setEditValue(newValue as Option);
+          setErrorMessage('');
+        }}
+        error={{ message: errorMessage, setMessage: setErrorMessage }}
+        isLoading={isLoading}
+        value={value}
+        format={format}
+        options={options}
+      />
+      {editOpen ? (
+        <div className={classes(css.buttonsDiv)}>
+          <CloseButton onClick={handleCancel} />
+          {errorMessage === '' && <ConfirmButton onClick={handleConfirm} />}
+        </div>
+      ) : (
+        <div className={classes(css.editButtonDiv)}>
+          <EditButton
+            fontSize="medium"
+            onClick={() => {
+              setEditValue(value);
+              setEditOpen(true);
+            }}
+          />
+        </div>
+      )}
+    </div>
+  );
+};
+
+export const EditableSelect: FC<
+  {
+    editOpen: boolean;
+    editValue: Option | undefined;
+    value: Option;
+    options: Option[];
+    isLoading: boolean;
+    format: string | undefined;
+    error: { message: string; setMessage: Dispatch<SetStateAction<string>> };
+  } & ComponentPropsWithoutRef<typeof Select>
+> = ({
+  editOpen,
+  editValue,
+  value,
+  options,
+  isLoading,
+  format,
+  error,
+  ...props
+}) => {
+  const [menuIsOpen, setMenuIsOpen] = useState(false);
+
+  useEffect(() => {
+    if (editOpen) setMenuIsOpen(true);
+    else setMenuIsOpen(false);
+  }, [editOpen]);
+
+  if (!editOpen)
+    return (
+      <div className={classes(css.value, css[format!])}>{value.label}</div>
+    );
+  if (isLoading) return <LoadingSpinner />;
+  return (
+    <>
+      <Select
+        {...props}
+        className={classes(css.editableSelect, css['react-select'])}
+        classNamePrefix="react-select"
+        styles={{
+          menuPortal: (base) => ({ ...base, zIndex: 9999 }),
+          indicatorsContainer: () => ({ paddingRight: '5% !important' }),
+        }}
+        value={editValue}
+        menuPortalTarget={document.body}
+        options={options}
+        isSearchable={false}
+        menuIsOpen={menuIsOpen}
+      />
+      <FieldError msg={error.message} />
```

Comment:
I'm not quite sure about this error display.

It's rendered to the side of the select, in a quite narrow space.
Also, the only error should happen when confirming the choice, but the select closes without showing any error.

Reply:
Good catch. I did not notice the styles.

At some point the error (in EditableText) showed request errors too, which was my intended use case. However, the page is reloaded whether the request fails or not. I suppose apiV2 patchTask invalidates Tasks-tag and reload happens due to tasks being fetched. It would be nice to invalidate tags only on succeeding requests, but I don't think it's possible... so currently errors is never shown in editableSelect. Not sure about what to do 🤔

Reply:
This works, great!